src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java

```diff
@@ -26,173 +27,209 @@
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageImpl;
 import org.springframework.data.domain.PageRequest;
+import org.zalando.problem.Problem;
 
-import se.sundsvall.incident.api.model.IncidentSaveRequest;
-import se.sundsvall.incident.dto.Category;
-import se.sundsvall.incident.dto.IncidentDto;
-import se.sundsvall.incident.dto.Status;
-import se.sundsvall.incident.integration.db.AttachmentRepository;
-import se.sundsvall.incident.integration.db.IncidentRepository;
+import se.sundsvall.incident.api.model.IncidentOepResponse;
+import se.sundsvall.incident.api.model.IncidentResponse;
+import se.sundsvall.incident.api.model.IncidentSaveResponse;
 import se.sundsvall.incident.integration.db.entity.IncidentEntity;
+import se.sundsvall.incident.integration.db.entity.util.Status;
+import se.sundsvall.incident.integration.db.repository.CategoryRepository;
+import se.sundsvall.incident.integration.db.repository.IncidentRepository;
 import se.sundsvall.incident.integration.lifebuoy.LifeBuoyIntegration;
 import se.sundsvall.incident.integration.messaging.MessagingIntegration;
 
+import generated.se.sundsvall.messaging.MessageResult;
+
 @ExtendWith(MockitoExtension.class)
 class IncidentServiceTest {
 
 	@Mock
-	private LifeBuoyIntegration lifeBuoyIntegration;
+	private LifeBuoyIntegration mockLifeBuoyIntegration;
 
 	@Mock
-	private MessagingIntegration messagingIntegration;
+	private MessagingIntegration mockMessagingIntegration;
 
 	@Mock
-	private IncidentRepository incidentRepository;
+	private CategoryRepository mockCategoryRepository;
 
 	@Mock
-	private AttachmentRepository attachmentRepository;
+	private IncidentRepository mockIncidentRepository;
 
 	@InjectMocks
 	private IncidentService incidentService;
 
 	@Test
-	void getOepIncidentStatus() {
-		final var incidentEntity = buildIncidentEntity(Category.VATTENMATARE);
-		when(incidentRepository.findIncidentEntityByExternalCaseId(anyString())).thenReturn(Optional.of(incidentEntity));
+	void fetchIncidentByIdTest() {
+		when(mockIncidentRepository.findById(anyString())).thenReturn(Optional.ofNullable(createIncidentEntity()));
+
+		var result = incidentService.fetchIncidentById(anyString());
+
+		assertThat(result).isNotNull().isInstanceOf(IncidentResponse.class);
+		verify(mockIncidentRepository, times(1)).findById(anyString());
```

Comment:
`times(1)` is redunant and can be removed

(Comment goes for all rows in class where `times(1)` is used)